qemu/tests/win_virtio_driver_installer_uninstall.py

```diff
@@ -62,19 +62,25 @@ def run(test, params, env):
                          expected_gagent_version)
     driver_check(session, test, params)
 
-    error_context.context("Run virtio-win-guest-tools.exe uninstall test",
+    uninstall_method = params.get('uninstall_method', 'installer')
+    error_context.context("Run uninstall test via %s" % uninstall_method,
                           test.log.info)
     vm.send_key('meta_l-d')
     time.sleep(30)
-    run_uninstall_cmd = utils_misc.set_winutils_letter(
-        session, params["run_uninstall_cmd"])
+    run_uninstall_cmd = utils_misc.set_winutils_letter(session, params["run_uninstall_cmd"])
     session.cmd(run_uninstall_cmd)
     time.sleep(30)
-    s_check, o_check = session.cmd_status_output(installer_pkg_check_cmd)
-    if s_check == 0:
-        test.fail("Could not uninstall Virtio-win-guest-tools package "
-                  "in guest', detail: '%s'" % o_check)
-    session = vm.reboot(session)
+    if uninstall_method == "msi":
+        check_warning_file = params["check_warning_file"]
+        output = session.cmd_output(check_warning_file)
+        if params["warning_message"] not in output:
+            test.fail("Not found expected warning message, the output is %s" % output)
```

Comment:
There is no reboot step but all the other check steps will be went through, do we know that they can all pass without reboot? Or maybe you want to add a `return` here to skip those steps?

Reply:
Updated the test steps, it the test run with msi, there is no need to check if the driver is uninstalled.Thanks!

 (1/2) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.q35: STARTED
 (1/2) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.q35: PASS (969.12 s)
 (2/2) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.win_virtio_driver_install_by_installer.driver_uninstall.by_msi.q35: STARTED
 (2/2) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.win_virtio_driver_install_by_installer.driver_uninstall.by_msi.q35: PASS (501.53 s)